File: NationalSecurityCommunity/app/src/main/java/com/national/security/community/injection/module/AppModule.java
package com.national.security.community.injection.module;


import com.national.security.community.App;
import com.national.security.community.data.ApiService;
import com.national.security.community.helper.RealmHelper;
import com.national.security.community.helper.RetrofitHelper;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;


@Module
public class AppModule {
    private final App application;

    public AppModule(App application) {
        this.application = application;
    }

    @Provides
    @Singleton
    App provideApplicationContext() {
        return application;
    }

    @Provides
    @Singleton
    RetrofitHelper provideRetrofitHelper(ApiService apiService) {
        return new RetrofitHelper(apiService);
    }

    @Provides
    @Singleton
    RealmHelper provideRealmHelper() {
        return new RealmHelper(application);
    }
}

File: app/src/main/java/com/jmd_news_kotlin/di/module/AppModule.kt
package com.jmd_news_kotlin.di.module

import com.jmd_news_kotlin.App
import com.jmd_news_kotlin.net.ApiService
import com.jmd_news_kotlin.utils.RetrofitHelper
import dagger.Module
import dagger.Provides
import retrofit2.Retrofit
import javax.inject.Singleton

@Module
class AppModule constructor(app: App) {

    private var application: App = app

    @Provides
    @Singleton
    fun provideApplicationContext(): App {
        return application
    }

    @Provides
    @Singleton
    fun providerApiService(retrofit: Retrofit): ApiService {
        return retrofit.create(ApiService::class.java)
    }

    @Provides
    @Singleton
    fun provideRetrofitHelper(apiService: ApiService): RetrofitHelper {
        return RetrofitHelper(apiService)
    }
}